code/modules/clothing/spacesuits/miscellaneous.dm

```diff
@@ -277,17 +277,25 @@ Contains:
 	actions_types = list()
 	item_flags = NODROP
 
-
 /obj/item/clothing/suit/space/hardsuit/carp
 	name = "carp space suit"
 	desc = "A slimming piece of dubious space carp technology, you suspect it won't stand up to hand-to-hand blows."
 	icon_state = "carp_suit"
 	item_state = "space_suit_syndicate"
 	slowdown = 0	//Space carp magic, never stop believing
 	armor = list("melee" = -20, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 100, "rad" = 75, "fire" = 60, "acid" = 75) //As whimpy whimpy whoo
-	allowed = list(/obj/item/tank/internals, /obj/item/gun/ballistic/automatic/speargun)	//I'm giving you a hint here
+	allowed = list(/obj/item/tank/internals, /obj/item/pneumatic_cannon/speargun)	//I'm giving you a hint here
 	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/carp
 
+/obj/item/clothing/head/helmet/space/hardsuit/carp/equipped(mob/living/carbon/human/user, slot)
+	..()
+	if (slot == SLOT_HEAD)
+		user.faction |= "carp"
+
+/obj/item/clothing/head/helmet/space/hardsuit/carp/dropped(mob/living/carbon/human/user)
+	..()
+	if (user.head == src)

```

Comment:
this proc is before it's removed right? If it handles removing and you call inheritance first, wouldn't this always be false?

Reply:
That doesn't seem to be the case; when I tested it, it removed and added the "carp" faction correctly.
I copied the code for it from the prototype (RD) hardsuit:
https://github.com/tgstation/tgstation/blob/a2e4460ba583612a939cf227e4d9ebbcf1c29f82/code/modules/clothing/spacesuits/hardsuit.dm#L466-L470

Reply:
kk I didn't do my homework at the time of OP